cloud-controller-manager/do/loadbalancers.go

```diff
@@ -1131,48 +1122,30 @@ func getStickySessionsCookieTTL(service *v1.Service) (int, error) {
 // getRedirectHTTPToHTTPS returns whether or not Http traffic should be redirected
 // to Https traffic for the loadbalancer. false is returned if not specified.
 func getRedirectHTTPToHTTPS(service *v1.Service) (bool, error) {
-	redirectHTTPToHTTPS, ok := service.Annotations[annDORedirectHTTPToHTTPS]
-	if !ok {
-		return false, nil
-	}
-
-	redirectHTTPToHTTPSBool, err := strconv.ParseBool(redirectHTTPToHTTPS)
+	redirectHTTPToHTTPS, _, err := getBool(service.Annotations, annDORedirectHTTPToHTTPS)
 	if err != nil {
-		return false, fmt.Errorf("failed to parse redirect HTTP-to-HTTPS flag %q from annotation %q: %s", redirectHTTPToHTTPS, annDORedirectHTTPToHTTPS, err)
+		return false, fmt.Errorf("failed to get HTTP-to-HTTPS configuration setting: %s", err)
```

Comment:
Why the change from `parse` to `get`?

Reply:
My thinking here is that parsing is just one aspect of getting a parameter: the value needs to be retrieved from the annotation map and then converted (with the latter being what I considered the essential parsing step).

Before this change, parsing was directly implemented by `getRedirectHTTPToHTTPS`, but now it's just an implementation detail of a more abstract function (`getBool`). Maintaining the wording to me means we'd be leaking implementation details into this function.

I guess you could make an argument that we could return `err` directly without wrapping. The rationale we do so for now is that it presumably makes the error description more readable by not just printing a rather technical sounding annotation name (`service.beta.kubernetes.io/do-loadbalancer-redirect-http-to-https`) but also a descriptive text (_HTTP-to-HTTPS_). We could discuss whether that's worth keeping, though I'd rather do that in a separate PR that updates all descriptions in one go.